### PR TITLE
MF-150: Add image selection actions

### PR DESCRIPTION
**Pull request by ivange94**
Show a list of actions when at least an image is
selected.

### CODE REVIEW

File: src/widgets/attachments/attachments-overview.component.tsx [hunk 1]
```diff
@@ -65,10 +67,52 @@ export default function AttachmentsOverview() {
     }
   }
 
+  function handleDragOver(e: React.SyntheticEvent) {
+    e.preventDefault();
+    e.stopPropagation();
+  }
+
   function handleCurrentImageChange(index) {
```
Comment (FlorianRappl) [inline src/widgets/attachments/attachments-overview.component.tsx]: `index: number`, right?

File: src/widgets/attachments/attachments-overview.component.tsx [hunk 2]
```diff
@@ -65,10 +67,52 @@ export default function AttachmentsOverview() {
...
     setCurrentImage(index);
   }
 
+  function handleImageSelect(index, image) {
```
Comment (FlorianRappl) [inline src/widgets/attachments/attachments-overview.component.tsx]: Is `image` used below? `index` is `number`, right?

File: src/widgets/attachments/attachments-overview.component.tsx [hunk 3]
```diff
@@ -65,10 +67,52 @@ export default function AttachmentsOverview() {
...
+    const result = Promise.all(
+      selected.map(att =>
+        deleteAttachment(att.id, abortController).then((response: any) => {})
+      )
+    );
+    result.then(() => {
+      const attachments_tmp = attachments.filter(
+        att => att.isSelected !== true
+      );
+      setAttachments(attachments_tmp);
```
Comment (FlorianRappl) [inline src/widgets/attachments/attachments-overview.component.tsx]: Dangerous - if we route to another component while this is in progress we'll get an error.

We need to cancel the promise (or not run the `setAttachments`) when the component unmounts.

File: src/widgets/attachments/attachments-overview.component.tsx [hunk 4]
```diff
@@ -105,6 +144,16 @@ export default function AttachmentsOverview() {
           />
         </form>
       </div>
+      {getSelectedImages().length !== 0 && (
+        <div className={styles.actions}>
+          <button
+            onClick={deleteSelected}
+            className={`omrs-btn omrs-filled-action`}
+          >
+            Delete Selected
```
Comment (FlorianRappl) [inline src/widgets/attachments/attachments-overview.component.tsx]: Translation required.

File: src/widgets/attachments/attachments-overview.component.tsx [hunk 5]
```diff
@@ -113,6 +162,7 @@ export default function AttachmentsOverview() {
             Delete
```
Comment (FlorianRappl) [inline src/widgets/attachments/attachments-overview.component.tsx]: I know this is existing code - sorry for being so picky (but I really want to have an eye on translations now); can you make it localizable, too?

File: src/widgets/attachments/attachments-overview.component.tsx (hunk 2 above)
Comment (ivange94) [reply src/widgets/attachments/attachments-overview.component.tsx]: Image is not used but that's the function signature required by the library.

Comment (FlorianRappl) [reply src/widgets/attachments/attachments-overview.component.tsx]: Okay, but then you can leave out `image`. Unused trailing arguments can always be removed.

**Comment by ivange94** [comment]
I am going to close this and make another PR. Some how I got other peoples commit in here.

**Comment by ivange94** [comment]
@FlorianRappl new PR with all concerns addressed https://github.com/openmrs/openmrs-esm-patient-chart-widgets/pull/85

**Pull request closed without merging**